compiler/ilgen/TypeDictionary.hpp

```diff
@@ -36,9 +36,9 @@
 #include "ilgen/IlBuilder.hpp"
 
 class TR_Memory;
-class TR_HashTabString;
 
 namespace OMR { class StructType; }
+namespace OMR { class UnionType; }
```

Comment:
I'm not super happy about these two type names showing up in the public header (I was trying to keep these types completely private), but I'll trade my happiness for the simplifying value of getStruct() and getUnion().